### ISSUE TITLE
Add a "share" feature to the Editor

### ISSUE THREAD

**Issue by laurenwalker** [NCEAS/metacatui#344]
We should expand the simple toggle from #343 into a more configurable "share" feature that gives the user the ability to add people and/or groups to specific permission lists.

We will probably want to make the wording more user-friendly and avoid language like "change permission", "write", and "access control." 

We may want to present a modal window where the user can type in the name of a person to search for an individual. The autocomplete feature will search the DataONE /accounts list. The user can select people or groups and check a checkbox next to their name to give them "edit" access (`write` permission). They can also add "dataset managers" or "owners", which would have `changePermission` access.

**Comment by laurenwalker** [comment]
I paved the way for this in the last commit here: 7cff77e02bcefd81e9106fe3d01c650fcb4e11db

Access Policies are parsed and serialized in separate functions in the DataONEObject model, which can moved to the AccessRule model and AccessPolicy collection. A skeleton for the AccessRule model was added in the commit above.

The AppModel defaultAccessPolicy should be parsed into an AccessRule model at the time of DataONEObject initialization.
